src/question/models.py

```diff
@@ -1,24 +1,56 @@
 from django.db import models
 from taggit.managers import TaggableManager
 from taggit.models import TagBase
-
-from common.models import AbstractComment, Votes
+from django.core.exceptions import ValidationError
+from common.models import AbstractComment
 from account.models import Profile
 
 
 class Question(AbstractComment):
     title = models.CharField(max_length=255)
     tags = TaggableManager()
-    parent = models.OneToOneField(Votes, related_name='question', on_delete=models.CASCADE, null=True)
+
+    def __str__(self):
+        return self.title
 
 
 class Answer(AbstractComment):
     is_accepted = models.BooleanField()
-    parent = models.OneToOneField(Votes, related_name='answer', on_delete=models.CASCADE, null=True)
+    parent = models.ForeignKey(
+        Question,
+        on_delete=models.CASCADE,
+        related_name='answers',
+        null=False
+    )
 
 
 class Comment(AbstractComment):
-    parent = models.OneToOneField(Votes, related_name='comment', on_delete=models.CASCADE, null=True)
+    parent_answer = models.ForeignKey(
+        'question.Answer',
+        on_delete=models.CASCADE,
+        related_name='comments',
+        null=True
+    )
+    parent_question = models.ForeignKey(
+        'question.Question',
+        on_delete=models.CASCADE,
+        related_name='comments',
+        null=True
+    )
+
+    @property
+    def parent(self):
+        return self.parent_answer or self.parent_question
+
+    def clean(self):
```

Comment:
The clean method is not invoked when calling a model's save method, only when saving using a ModelForm. To make this validation work, override the save method and call clean or full_clean there.

Reply:
Probably fixed in https://github.com/DevTeamSCH/vikoverflow-backend/pull/11/commits/c70fc16f0b3b8ae26850087c89ea6c1c74c87454, not tested yet.

Reply:
The clean method is never invoking in DRF. (We should call it from the validate function of the serializer.)

Reply:
https://www.django-rest-framework.org/api-guide/validators/#validation-in-rest-framework

"With ModelForm the validation is performed partially on the form, and partially on the model instance. With REST framework the validation is performed entirely on the serializer class. "

Reply:
Fixed in https://github.com/DevTeamSCH/vikoverflow-backend/pull/11/commits/ed7db1f20d6972626a820013a43f4200bfadf9e2.